<a href="https://colab.research.google.com/github/novaRasyadina/Checkpoint1_avd_Nova-Rasyadina/blob/main/Checkpoint2_AVD_Nova_Rasyadina_A_2409116092.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Data Understanding**

# **Sumber Data**

Dataset yang digunakan pada analisis ini adalah "Games and Success in Students". data ini bersumber dari Kaggle yang berisi informasi kode sekolah, intensitas bermain game, dan capaian akademik siswa.

Sumber Dataset: https://www.kaggle.com/datasets/deadier/play-games-and-success-in-students

# **Struktur data**

Berikut merupakan struktur data dari database ini:

*   Sex (jenis kelamin) : 0 = Perempuan, 1 = Laki-laki
*   School Code : Kode Sekolah
*   Playing Years : Jumlah tahun bermain Game
*   Playing Often : intensitas keseringan bermain game dengan skala 0-5
*   Playing Hours : Rata-rata jam bermain game per hari
*   Playing Game : Jumlah jenis game yang dimainkan dengan skala 0-2
*   Parent Revenue : Pendapatan orang tua dengan skala 0-4
*   Father Education : Pendidikan Ayah dengan skala 0-6
*   Mother Education : Pendidikan ibu dengan skala 0-6
*   Grade : capaian Akademik/Nilai siswa







# **Informasi Data**

In [1]:
import pandas as pd

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
df = pd.read_csv("/content/drive/MyDrive/ADV/gameandgrade.csv")
print(df)

     Sex  School Code  Playing Years  Playing Often  Playing Hours  \
0      0            1              1              2              1   
1      1            1              1              3              1   
2      0            1              0              0              0   
3      0            1              3              5              1   
4      1            1              1              1              2   
..   ...          ...            ...            ...            ...   
765    1            2              1              2              4   
766    1            9              2              5              1   
767    0            4              2              1              3   
768    0            4              2              1              3   
769    0            9              2              5              2   

     Playing Games  Parent Revenue  Father Education  Mother Education Grade  
0                1               4                 4                 5  77.5  
1

# **Melihat Missing Value pada dataset**

In [4]:
print((df.isna().sum() / len(df)) * 100)

Sex                 0.0
School Code         0.0
Playing Years       0.0
Playing Often       0.0
Playing Hours       0.0
Playing Games       0.0
Parent Revenue      0.0
Father Education    0.0
Mother Education    0.0
Grade               0.0
dtype: float64


dari hasil analisis tidak ditemukan missing value/nilai hilang

# **Duplicated Values**

kondisi dimana ada data yang muncul beberapa kali dalam satu data set.

In [17]:
df[df.duplicated()]

Sex  School Code  Playing Years  Playing Often  Playing Hours  \
278    0            4              0              0              0   
367    0            4              0              0              0   
419    0            5              0              0              0   
468    0            7              0              0              0   
478    0            7              0              0              0   
480    0            7              0              0              0   
610    0            9              0              0              0   
616    0            9              0              0              0   
617    0            9              0              0              0   
618    0            9              0              0              0   
619    0            9              0              0              0   
631    0            9              0              0              0   
640    0            9              0              0              0   
644    0            9              0              0              0   
646    0            9              0              0              0   
746    0            8              0              0              0   

     Playing Games  Parent Revenue  Father Education  Mother Education Grade  
278              0               2                 3                 4    70  
367              0               1                 3                 2    70  
419              0               2                 5                 4   100  
468              0               0                 4                 4   100  
478              0               0                 5                 5   100  
480              0               0                 6                 6   100  
610              0               0                 5                 5   100  
616              0               0                 6                 6   100  
617              0               0                 5                 6   100  
618              0               0                 6                 6   100  
619              0               0                 5                 5   100  
631              0               0                 5                 5   100  
640              0               0                 4                 4   100  
644              0               0                 4                 4   100  
646              0               0                 4                 4   100  
746              0               2                 3                 3    75

Karena dataset ini adalah sebuah survey, jadi terdapat banyak kemunculan data yang sama, maka dari itu tidak perlu melakukan drop atau menghapus data yang sama

# **Outliers**

Melakukan pengecekan outliers

In [18]:
results = []

cols = df.select_dtypes(include=['float64', 'int64'])

for col in cols:
  q1 = df[col].quantile(0.25)
  q3 = df[col].quantile(0.75)
  iqr = q3 - q1
  lower_bound = q1 - 1.5*iqr
  upper_bound = q3 + 1.5*iqr
  outliers = df[(df[col] < lower_bound) | (df[col] > upper_bound)]
  percent_outliers = (len(outliers)/len(df))*100
  results.append({'Kolom': col, 'Persentase Outliers': percent_outliers})

results_df = pd.DataFrame(results)
results_df.set_index('Kolom', inplace=True)
results_df = results_df.rename_axis(None, axis=0).rename_axis('Kolom', axis=1)

display(results_df)

Kolom             Persentase Outliers
Sex                          0.000000
School Code                  0.000000
Playing Years                0.000000
Playing Often                0.000000
Playing Hours                0.000000
Playing Games                0.000000
Parent Revenue               6.493506
Father Education             0.000000
Mother Education             0.000000

hanya Parent Revenue yang memiliki outliers sekitar 6.49%, Kolom lainnya tidak memiliki outliers yang signifikan.
karena ini merupakan data survey, outliers tidak perlu dihapus, karena Data skala biasanya bersifat subjektif, jadi meskipun ada nilai ekstrem, itu masih bisa mencerminkan persepsi responden.
Menghapus outliers bisa menghilangkan informasi penting, terutama jika ada kelompok responden dengan pandangan atau kondisi yang berbeda.

# **Inconsistent Values**

In [19]:
df

Sex  School Code  Playing Years  Playing Often  Playing Hours  \
0      0            1              1              2              1   
1      1            1              1              3              1   
2      0            1              0              0              0   
3      0            1              3              5              1   
4      1            1              1              1              2   
..   ...          ...            ...            ...            ...   
765    1            2              1              2              4   
766    1            9              2              5              1   
767    0            4              2              1              3   
768    0            4              2              1              3   
769    0            9              2              5              2   

     Playing Games  Parent Revenue  Father Education  Mother Education Grade  
0                1               4                 4                 5  77.5  
1                1               1                 3                 3    83  
2                0               1                 3                 3    80  
3                1               2                 2                 3    45  
4                1               1                 3                 4    85  
..             ...             ...               ...               ...   ...  
765              1               1                 3                 3  67.5  
766              1               2                 5                 5   100  
767              1               4                 5                 4    75  
768              1               2                 4                 4    77  
769              1               0                 5                 5   100  

[770 rows x 10 columns]

# **Pengecekan nilai tidak konsisten**

Berdasarkan hasil pengecekan sebelumnya semua kolom memiliki nilai dalam rentang yang wajar berdasarkan skala survei.
Tidak ditemukan nilai aneh atau tidak konsisten dalam dataset, dan hal ini kembali lagi pada bentuk dataset yang berupa survey dari responsen yang bisa saja memiliki respon sama dalam rentang skala

In [25]:
for col in ["Sex", "School Code", "Playing Games"]:
    df[col] = df[col].fillna('')
    df[col] = df[col].str.lower().str.strip()
    df[col] = df[col].str.replace(r'[^a-z0-9\s]', '', regex=True)
    df[col] = df[col].str.title()

In [26]:
df

Sex School Code  Playing Years  Playing Often  Playing Hours  \
0     0           1              1              2              1   
1     1           1              1              3              1   
2     0           1              0              0              0   
3     0           1              3              5              1   
4     1           1              1              1              2   
..   ..         ...            ...            ...            ...   
765   1           2              1              2              4   
766   1           9              2              5              1   
767   0           4              2              1              3   
768   0           4              2              1              3   
769   0           9              2              5              2   

    Playing Games  Parent Revenue  Father Education  Mother Education Grade  
0               1               4                 4                 5  77.5  
1               1               1                 3                 3    83  
2               0               1                 3                 3    80  
3               1               2                 2                 3    45  
4               1               1                 3                 4    85  
..            ...             ...               ...               ...   ...  
765             1               1                 3                 3  67.5  
766             1               2                 5                 5   100  
767             1               4                 5                 4    75  
768             1               2                 4                 4    77  
769             1               0                 5                 5   100  

[770 rows x 10 columns]

# **Data Construction**

Karena analisis ini untuk mengetahui hubungan bermain game dengan capaian akademik siswa, jadi data construction bisa digunakan untuk membuat fitur baru seperti otal waktu bermain per minggu, kategori gamer, atau rata-rata pendidikan orang tua.

In [27]:
df["Total Gaming Hours Per Week"] = df["Playing Often"] * df["Playing Hours"]

In [28]:
def categorize_gamer(hours):
    if hours == 0:
        return "Non-Gamer"
    elif hours <= 5:
        return "Casual"
    elif hours <= 15:
        return "Regular"
    else:
        return "Hardcore"

df["Gamer Category"] = df["Total Gaming Hours Per Week"].apply(categorize_gamer)

In [29]:
df["Parent Education Avg"] = (df["Father Education"] + df["Mother Education"]) / 2

In [33]:
print(df.head())

  Sex School Code  Playing Years  Playing Often  Playing Hours Playing Games  \
0   0           1              1              2              1             1   
1   1           1              1              3              1             1   
2   0           1              0              0              0             0   
3   0           1              3              5              1             1   
4   1           1              1              1              2             1   

   Parent Revenue  Father Education  Mother Education Grade  \
0               4                 4                 5  77.5   
1               1                 3                 3    83   
2               1                 3                 3    80   
3               2                 2                 3    45   
4               1                 3                 4    85   

   Total Gaming Hours Per Week Gamer Category  Parent Education Avg  
0                            2         Casual                   4.5  


# **Data Reduction**

In [35]:
def categorize_revenue(revenue):
    if revenue <= 2:
        return "Low"
    elif revenue <= 4:
        return "Medium"
    else:
        return "High"

df["Parent Revenue Category"] = df["Parent Revenue"].apply(categorize_revenue)

In [36]:
df.drop(columns=["Parent Revenue"], inplace=True)

In [37]:
df.head()

Sex School Code  Playing Years  Playing Often  Playing Hours Playing Games  \
0   0           1              1              2              1             1   
1   1           1              1              3              1             1   
2   0           1              0              0              0             0   
3   0           1              3              5              1             1   
4   1           1              1              1              2             1   

   Father Education  Mother Education Grade  Total Gaming Hours Per Week  \
0                 4                 5  77.5                            2   
1                 3                 3    83                            3   
2                 3                 3    80                            0   
3                 2                 3    45                            5   
4                 3                 4    85                            2   

  Gamer Category  Parent Education Avg Parent Revenue Category  
0         Casual                   4.5                  Medium  
1         Casual                   3.0                     Low  
2      Non-Gamer                   3.0                     Low  
3         Casual                   2.5                     Low  
4         Casual                   3.5                     Low